include/dice/template-library/flex_array.hpp

```diff
@@ -8,11 +8,22 @@
 #include <span>
 #include <stdexcept>
 
+#if __has_include(<ankerl/svector.h>)
 #include <ankerl/svector.h>
+#endif // __has_include
 
 namespace dice::template_library {
     using std::dynamic_extent;
 
+	/**
+	 * The underlying implementation of a flex array
+	 */
+	enum struct flex_array_implementation {
+		array, ///< max and current size determined statically
+		array_and_size, ///< max size determined statically, current size determined dynamically
+		sbo_vector, ///< small buffer optimized vector
+	};
+
 	namespace detail_flex_array {
 		template<typename T, size_t extent, size_t max_extent>
```

Comment:
```suggestion
		template<typename T, size_t extent, size_t capacity>
```

How about using capacity here instead of max_extend

Reply:
`capacity` is inaccurate because `capacity` in C++ usually means: the amount of memory that can be accessed without reallocation

Reply:
and in the case of `max_extent == dynamic_extent` (svector impl) this makes intuitive sense I think

Reply:
makes sense